extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1174,11 +1178,27 @@ export class AmpA4A extends AMP.BaseElement {
             'src': srcPath,
             'name': nameData,
           }, SHARED_IFRAME_PROPERTIES));
+      if (method == XORIGIN_MODE.NAMEFRAME) {
+        const attributes = this.generateSentinelAndContext(iframe);
+        attributes['creative'] = creative;
+        const name = JSON.stringify(attributes);
+        iframe.setAttribute('name', name);
+      }
       return this.iframeRenderHelper_(iframe);
     });
   }
 
   /**
+   * Generates sentinel for iframe and gets the context metadata.
+   * @param {!Element} iframe
+   * @return {!Object} context
```

Comment:
It's generally not advisable to use `Object` as a type parameter, but here it's derived directly from a type signature in other code and I can't blame you for not chasing down and fixing the other code since this needs to get merged soon.